src/OpenSpaceToolkit/Astrodynamics/Dynamics/Tabulated.cpp

```diff
@@ -0,0 +1,139 @@
+/// Apache License 2.0
+
+#include <OpenSpaceToolkit/Core/Error.hpp>
+#include <OpenSpaceToolkit/Core/Types/Index.hpp>
+
+#include <OpenSpaceToolkit/Astrodynamics/Dynamics/Tabulated.hpp>
+
+namespace ostk
+{
+namespace astro
+{
+namespace dynamics
+{
+
+using ostk::core::types::Index;
+
+using ostk::math::object::VectorXd;
+
+Tabulated::Tabulated(
+    const Array<Instant>& anInstantArray,
+    const MatrixXd& aContributionProfile,
+    const Array<Shared<const CoordinatesSubset>>& aWriteCoordinatesSubsets
+)
+    : Dynamics("Tabulated"),
+      contributionProfile_(aContributionProfile),
+      instants_(anInstantArray),
+      writeCoordinatesSubsets_(aWriteCoordinatesSubsets)
+{
+    if (anInstantArray.getSize() != (Index)aContributionProfile.rows())
+    {
+        throw ostk::core::error::RuntimeError(
+            "Contribution profile must have the same number of rows as the number of instants."
+        );
+    }
+
+    if (aWriteCoordinatesSubsets
+            .map<Index>(
+                [](const auto& coordinatesSubset)
+                {
+                    return coordinatesSubset->getSize();
+                }
+            )
+            .reduce(std::plus<Index>()) != (Index)aContributionProfile.cols())
+    {
+        throw ostk::core::error::RuntimeError(
+            "Contribution profile must have the same number of columns as the sum of the sizes of the write "
+            "coordinates subsets."
+        );
+    }
+
```

Comment:
Upon further inspection, it seems that we are propagating fine with larger step sizes (atleast for the constant thrust). IMO it's kind of arbitrary to force a step size, as it might vary from model to model. For now I think we can allow it. We can revisit this in the future.

Reply:
OK